packages/modular-scripts/src/cli.ts

```diff
@@ -343,6 +339,44 @@ async function build(directoryName: string, preserveModules?: boolean) {
   }
 }
 
+interface NPMRegistryListing {
+  "dist-tags": Record<string, string>;
+}
+
+async function preflight() {
+  if (process.env.SKIP_PREFLIGHT_CHECK !== 'true') {
+    const { stdout: registry } = await execa('yarnpkg', ['config', 'get', 'registry']);
+
+    const url = String(new URL("/modular-scripts", registry));
+
+    const res = await fetch(url);
+    const modualarScriptsRegistry: NPMRegistryListing = await res.json();
+    const newVersion = modualarScriptsRegistry["dist-tags"]["latest"];
+
+    const { version }: ModularPackageJson = fs.readJSONSync(path.join(__dirname, "..", "package.json"));
```

Comment:
This doesn't seem right, this will only work in our own repository (which is... odd, what does updating mean then). Did you mean to use dependencies['modular-scripts'] and compare with `semver` instead?

Also, instead of using package.json, maybe you should be using yarn.lock instead?

Reply:
Oh it'll look at the package.json of itself. I think you should use find-up to find the nearest package.json instead, because you can't guarantee that this file will be at this path (especially after we migrate to using `modular build` to build modular-scripts)

Reply:
If I can work out jest mocks then it should be easy enough to mock out the `fetch` and the `fse` call to return differing things, then just assert that prompts was called?

Reply:
Mocking fetch is a dope idea. Yeah I think you're right, that should do it.

Reply:
I went around in circles a couple of times around this. There's also the question of how we deal with a major releases (once we get to `1.0.0`) and make that transition easy for people - although this preflight would give us the option to run codemods on install as well automatically.